numpy/lib/tests/test__iotools.py

```diff
@@ -212,6 +213,7 @@ def test_upgrademapper(self):
         assert_equal(test, date(2009, 1, 1))
         test = convert('')
         assert_equal(test, date(2000, 1, 1))
+        StringConverter._mapper = _original_mapper
```

Comment:
This should be in a `try` / `finally` , in case an assertion fails

Reply:
(is this change related to the patch as a whole?

Reply:
This keeps adding to `_mapper` which is a list and thus grows. So it trips `pytest-leaks` because the total reference count increases each time the test runs.

Reply:
Got it, thanks